python/pyarrow/_cuda.pyx

```diff
@@ -43,9 +45,57 @@ cdef class Context:
             self.context.reset()
             raise ValueError('device_number argument must be '
                              'non-negative less than %s' % (n))
-        check_status(manager.GetContext(device_number, &self.context))
+        if handle == 0:
+            check_status(manager.GetContext(device_number, &self.context))
+        else:
+            check_status(manager.CreateSharedContext(device_number,
+                                                     <void*>handle,
+                                                     &self.context))
         self.device_number = device_number
 
+    @staticmethod
+    def from_numba(context=None):
+        """Create Context instance from a numba CUDA context.
+
+        Parameters
+        ----------
+        context : {numba.cuda.cudadrv.driver.Context, None}
+          Specify numba CUDA context instance. When None, use the
+          current numba context.
+
+        Returns
+        -------
+        shared_context : pyarrow.cuda.Context
+          Context instance.
+        """
+        if context is None:
+            import numba.cuda
+            context = numba.cuda.cudadrv.devices.get_context()
```

Comment:
Can [numba.cuda.current_context](https://numba.pydata.org/numba-doc/dev/cuda-reference/host.html#numba.cuda.current_context) be used instead?

Reply:
Yes. Fixed.